lib/vagrant-openshift/helper/command_helper.rb

```diff
@@ -141,56 +141,72 @@ def sync_bash_command_on_dockerfile(repo_name, dockerfile_build_path, build_cmd)
         sync_bash_command(repo_name, build_cmd, file_path, branch_name, sync_path, commit_id_path)
       end
 
+      def repo_clone_bash_command(repo, url)
+        command = %{
+cloned=false
+echo 'Cloning #{repo} ...'
+}
+        if @options[:user]
+          user_repo_url="git@github.com:#{@options[:user]}/#{repo}"
+          command += %{
+echo 'Cloning #{user_repo_url}'
+git clone --quiet --recurse-submodules #{user_repo_url}
+if [ $? -eq 0 ]; then
+cloned=true
+(cd #{repo} && git remote add upstream #{url} && git fetch upstream)
+else
+echo 'Fork of repo #{repo} not found. Cloning read-only copy from upstream'
+fi
+}
+        end
+        command += %{
+[ $cloned != true ] && git clone --quiet --recurse-submodules #{url}
+(cd #{repo} && git checkout #{branch} &>/dev/null)
+}
+        command
+      end
+
       def repo_checkout_bash_command(repo, url)
         repo_path = File.expand_path(repo)
         command = ""
         branch = @options[:branch] || 'master'
+        check_and_fix_repo(repo_path)
```

Comment:
In general this is extremely hard to read.  The logic is extremely broken between ruby and bash.